core/modules/layout/layout.module

```diff
@@ -601,6 +601,7 @@ function layout_autoload_info() {
     'Block' => 'includes/block.class.inc',
     'BlockBroken' => 'includes/block.class.inc',
     'BlockLegacy' => 'includes/block_legacy.class.inc',
+    'TextBlock' => 'includes/block.text.inc',
```

Comment:
I feel like we should keep this under the `Block` namespace (not 100% on that) and name this `BlockText` instead of `TextBlock`. TextBlock would be a block owned by text.module. Of course, Layout module is hijacking Block's namespaces here, but I think that's okay.